src/cli_plugin/install/opensearch_dashboards.js

```diff
@@ -38,10 +38,11 @@ import { versionSatisfies, cleanVersion } from '../../legacy/utils/version';
 
 export function existingInstall(settings, logger) {
   try {
-    statSync(path.join(settings.pluginDir, kebabCase(settings.plugins[0].id)));
```

Comment:
You may have been right to worry that previous versions of plugins may have been installed into `settings.plugins[0].id` (1.1 installation on top of 1.0). Is that a valid concern. If so, make an array of all possible folder names, and do a foreach on them to check.

Reply:
One of the cases I can think of now is the user first used the id of the plugin to install it, then added a folderName property to `opensearch_dashboards.json` and tried installing again. It should fail saying the plugin is already installed. I will add both the checks to be on the safer side.